policy-controller/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM --platform=$BUILDPLATFORM ghcr.io/linkerd/dev:v39-rust-musl as controller
```

Comment:
This version is out of sync with the rest of the repo. That is *supposed* to be a blocker -- i.e. we're supposed to prevent this in CI. Our checks may not be rigorous enough to catch all dockerfiles, though.

There's an open PR (#9934) that started updating, but went too far into updating tooling so that our arm environment will not work... There's probably a simpler update path.